internal/credential/vault/jobs.go

```diff
@@ -0,0 +1,200 @@
+package vault
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/db/timestamp"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/kms"
+	"github.com/hashicorp/boundary/internal/scheduler"
+	"github.com/hashicorp/go-hclog"
+	vault "github.com/hashicorp/vault/api"
+)
+
+const (
+	defaultTokenRenewalInterval = time.Minute * 5
+	tokenRenewalWindow          = 10 // in minutes
+)
+
+type TokenRenewalJob struct {
+	repoFn RepositoryFactory
```

Comment:
Ill chat with @mgaffney about this tomorrow

Reply:
Since the repo can be used concurrently (its running while another service looks at `NextRunTime`) its considered not safe and therefore we do need to use repoFn